components/client/src/main/java/com/hotels/styx/client/connectionpool/stubs/StubConnectionFactory.java

```diff
@@ -56,6 +58,11 @@ public StubConnection(Origin origin) {
             return operation.get();
         }
 
+        @Override
+        public Observable<HttpResponse> write(HttpRequest request) {
+            throw new IllegalStateException("Not implemented");
```

Comment:
`java.lang.UnsupportedOperationException` is the correct exception for this scenario.

Reply:
Done.